drivers/meter/ade7758/ade7758.h

```diff
@@ -0,0 +1,515 @@
+#define ADE7758_SAGC_MSK					NO_OS_BIT(5)
+#define ADE7758_SAGB_MSK					NO_OS_BIT(4)
+#define ADE7758_SAGA_MSK					NO_OS_BIT(3)
+#define ADE7758_VAEHF_MSK					NO_OS_BIT(2)
+#define ADE7758_REHF_MSK					NO_OS_BIT(1)
+#define ADE7758_AEHF_MSK					NO_OS_BIT(0)
+
+/* ADE7758_REG_GAIN Bit Definition */
+#define ADE7758_INTEGRATOR_EN_MSK				NO_OS_BIT(7)
+#define ADE7758_PGA2_GAIN_MSK					NO_OS_GENMASK(6, 5)
+#define ADE7758_FULL_SCALE_MSK					NO_OS_GENMASK(4, 3)
+#define ADE7758_PGA1_GAIN_MSK					NO_OS_GENMASK(1, 0)
+
+/**
+ * @enum ade7758_dismod_e
+ * @brief ADE7758 ADCs can be turned off by using these bits.
+ */
+enum ade7758_dismod_e {
+	/* normal operation */
+	ADE7758_NORMAL_OPERATION,
+	/* switch off the current channel ADCs*/
+	ADE7758_SWITCH_OFF_CURRENT_ADC,
+	/* switch off the voltage channel ADCs */
+	ADE7758_SWITCH_OFF_VOLTAGE_ADC,
+	/* put the ADE7758 in sleep mode */
+	ADE7758_SLEEP_MODE,
+	/* swap voltage and currents channels */
+	ADE7758_SWAP_VOLTAGE_CURRENT,
+	/* switch off current channel ADCsand redirect the current
+	input signal to the voltage channel signal paths*/
+	ADE7758_OFF_CURRENT_ADCS_REDIRECTED_TO_VOLTAGE,
+	/* switch off voltage channel ADCs and redirect the voltage
+	input signal to the current channel signal paths */
+	ADE7758_OFF_VOLTAGE_ADCS_REDIRECTED_TO_CURRENT,
+	/* put the ADE7758 in power-down mode*/
+	ADE7758_POWER_DOWN_MODE
+};
+
+/**
+ * @enum ade7758_phsel_e
+ * @brief ADE7758 select the phase of the waveform sample or
+ * the source for the voltage line frequency measurement.
+ */
+enum ade7758_phsel_e {
+	/* phase A */
+	ADE7758_PHASE_A,
+	/* phase B */
+	ADE7758_PHASE_B,
+	/* phase C */
+	ADE7758_PHASE_C
+};
+
+/**
+ * @enum ade7758_wavesel_e
+ * @brief ADE7758 select the type of waveform.
+ */
+enum ade7758_wavesel_e {
+	/* Current */
+	ADE7758_CURRENT,
+	/* Voltage */
+	ADE7758_VOLTAGE,
+	/* Active power */
+	ADE7758_ACTIVE_POWER,
+	/* Reactive power */
+	ADE7758_REACTIVE_POWER,
+	/* Apparent power */
+	ADE7758_APPARENT_POWER
+};
+
+/**
+ * @enum ade7758_data_rate_e
+ * @brief ADE7758 select the data rate.
+ */
+enum ade7758_data_rate_e {
+	/* 26.04 kSPS (CLKIN/3/128) */
+	ADE7758_DATA_RATE_26_04KSPS,
+	/* 13.02 kSPS (CLKIN/3/256) */
+	ADE7758_DATA_RATE_13_02KSPS,
+	/* 6.51 kSPS (CLKIN/3/512) */
+	ADE7758_DATA_RATE_6_51KSPS,
+	/* 3.25 kSPS (CLKIN/3/1024) */
+	ADE7758_DATA_RATE_3_25KSPS
+};
+
+/**
+ * @enum ade7758_consel_e
+ * @brief ADE7758 select the input to the energy accumulation registers.
+ */
+enum ade7758_consel_e {
+	/* Consel 0 */
+	ADE7758_CONSEL_0,
+	/* Consel 1 */
+	ADE7758_CONSEL_1,
+	/* Consel 2 */
+	ADE7758_CONSEL_2
+};
+
+/**
+ * @enum ade7758_pga_gain_e
+ * @brief ADE7758 select the PGA gain.
+ */
+enum ade7758_pga_gain_e {
+	/* Gain */
+	/* gain 1 */
+	ADE7758_PGA_GAIN_1,
+	/* gain 2 */
+	ADE7758_PGA_GAIN_2,
+};
+
+/**
+ * @enum ade7758_ch_full_scale_sel_e
+ * @brief ADE7758 Current input full-scale select.
+ */
+enum ade7758_ch_full_scale_sel_e {
```

Comment:
having _e suffix is nice probably but inconsistent with the rest of no-OS code so idk...